repair/repair.cc

```diff
@@ -1153,34 +1153,34 @@ future<int> repair_service::do_repair_start(sstring keyspace, std::unordered_map
 
         if (nr_tablet_table != 0) {
             if (nr_vnode_table != 0) {
-                throw std::runtime_error("Mixed vnode table and tablet table");
+                throw std::invalid_argument("Mixed vnode table and tablet table");
             }
             is_tablet = true;
         }
         if (is_tablet) {
             // Reject unsupported options for tablet repair
             if (!options.start_token.empty()) {
-                throw std::runtime_error("The startToken option is not supported for tablet repair");
+                throw std::invalid_argument("The startToken option is not supported for tablet repair");
             }
             if (!options.end_token.empty()) {
-                throw std::runtime_error("The endToken option is not supported for tablet repair");
+                throw std::invalid_argument("The endToken option is not supported for tablet repair");
             }
             if (options.small_table_optimization) {
-                throw std::runtime_error("The small_table_optimization option is not supported for tablet repair");
+                throw std::invalid_argument("The small_table_optimization option is not supported for tablet repair");
             }
 
             // Reject unsupported option combinations.
             if (!options.data_centers.empty() && !options.hosts.empty()) {
-                throw std::runtime_error("Cannot combine dataCenters and hosts options.");
+                throw std::invalid_argument("Cannot combine dataCenters and hosts options.");
             }
             if (!options.ignore_nodes.empty() && !options.hosts.empty()) {
-                throw std::runtime_error("Cannot combine ignore_nodes and hosts options.");
+                throw std::invalid_argument("Cannot combine ignore_nodes and hosts options.");
             }
 
             auto host2ip = [&addr_map = _addr_map] (locator::host_id host) -> future<gms::inet_address> {
                 auto ip = addr_map.local().find(raft::server_id(host.uuid()));
                 if (!ip) {
-                    throw std::runtime_error(format("Could not get ip address for host {} from raft_address_map", host));
+                    throw std::invalid_argument(format("Could not get ip address for host {} from raft_address_map", host));
```

Comment:
This is not something user provided as a parameter.  Better to skip it.

Reply:
the `hosts` are specified by user. and we use `host2ip` to map them to their IP addresses. if we cannot find any of them, there are two explanations:

- one ore more of the hosts provided by `options.hosts` are invalid
- we just fail to keep track of the valid host to IP mapping

so, i think there are chances that it is provided by user.

Reply:
The host2ip is used as:

```
            for (auto& r : m.replicas) {
                auto shard = r.shard;
                auto ip = co_await host2ip(r.host);
```

the hostid is not provided by user. It is from the tablet meta data.

Also, it converts from host id to ip address. It has nothing to do with the optionts.hosts.

Reply:
my fault. will revert this part. thanks for pointing this out, Asias.